frontend/packages/dev-console/src/components/pipelines/pipeline-topology/TaskListNode.tsx

```diff
@@ -82,7 +84,7 @@ const TaskListNode: React.FC<TaskListNodeProps> = ({ element, unselectedText })
           <div className="pf-c-dropdown pf-m-expanded">
             <ul className="pf-c-dropdown__menu pf-m-align-right oc-kebab__popper-items odc-task-list-node__list-items">
               {options.map((option) => (
-                <li key={option.label}>
+                <li key={`${option.label}-${option.icon?.['props']?.kind}`}>
```

Comment:
Not a fan of digging into props unless you absolutely have to... I think since we have a little sandbox of code here, we could change things up a bit with the help of TypeScript.

```
type KeyedKebabOption = KebabOption & { key: string };

const taskToOption = (task: PipelineResourceTask, callback: NewTaskNodeCallback): KeyedKebabOption => {
```

That way we return a `option.key` value that we can use. You can set `key` simply to `${kind}-${name}` with the data.